cmd/analyzer/main.go

```diff
@@ -101,25 +106,57 @@ func analyzer() error {
 			return cmd.FailErr(err, "create docker client")
 		}
 		img, err = local.NewImage(
-			repoName,
+			imageName,
 			dockerClient,
-			local.FromBaseImage(repoName),
+			local.FromBaseImage(imageName),
 		)
 		if err != nil {
 			return cmd.FailErr(err, "access previous image")
 		}
 	} else {
 		img, err = remote.NewImage(
-			repoName,
+			imageName,
 			auth.EnvKeychain(cmd.EnvRegistryAuth),
-			remote.FromBaseImage(repoName),
+			remote.FromBaseImage(imageName),
 		)
 		if err != nil {
 			return cmd.FailErr(err, "access previous image")
 		}
 	}
 
-	md, err := analyzer.Analyze(img)
+	var cacheStore lifecycle.Cache
+	if cacheImageTag != "" {
+		origCacheImage, err := remote.NewImage(
+			cacheImageTag,
+			auth.EnvKeychain(cmd.EnvRegistryAuth),
+			remote.FromBaseImage(cacheImageTag),
+		)
+		if err != nil {
+			return cmd.FailErr(err, "accessing cache image")
+		}
+
+		emptyImage, err := remote.NewImage(
+			cacheImageTag,
+			auth.EnvKeychain(cmd.EnvRegistryAuth),
+			remote.WithPreviousImage(cacheImageTag),
+		)
+		if err != nil {
+			return cmd.FailErr(err, "creating new cache image")
+		}
+
+		cacheStore = cache.NewImageCache(
+			origCacheImage,
+			emptyImage,
+		)
+	} else {
```

Comment:
Similar to the exporter/cacher PR, I think we need to handle the case where there is no cache. When we check args in the `main` func we should warn if both `cacheDir`and `cacheImage` are empty and provide a noop cache to the analyzer.

Reply:
Added a warning and modified analyzer to use empty cache metadata.